extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -115,6 +125,13 @@ export class AmpAutocomplete extends AMP.BaseElement {
         `${TAG} should contain exactly one <input> child`);
     this.inputElement_ = inputElements[0];
 
+    if (this.templates_.hasTemplate(
+        this.element, 'template, script[template]')) {
+      this.templateElement_ =
+        this.templates_.findTemplate(this.element,
```

Comment:
ditto above, also we can simplify this to
```
const template = this.templates_.findTemplate(..
if( template ) { ... }
```

Reply:
The reason the `hasTemplate` check is here is because the `findTemplate` call will `userAssert` that one exists. Since the component does not require there is a template in all cases, we don't want to error in situations where the developer does not need or want a template. I see your point that the two calls is undesirable, so ideally there is a more suitable condition where the error is expected if the template can't be found. 

I'd suggest something like `if typeof this.inlineData_ is Array<!JsonObject>`, but is it possible to check a type with that much specificity? I only know the `isArray` and `instanceof`, but not a way to know it is an array of items of uniform type without looping through each item. Also, the `inlineData` has not yet been retrieved by the time the `buildCallback` begins to look for the template in the buildCallback, so it is hard to know when we would need a template.

Hypothetically also the developer _could_ provide a template for even plain text data, if they wanted to stylize it a certain way without needing additional data on a per-item basis, so my last point actually would not be an exhaustive condition.

Reply:
I see! We could add ` findtemplateOrNull` but this is totally fine, it's not a hot code path.